src/transforms/babel/babelrc.js

```diff
@@ -0,0 +1,219 @@
+const CORE_DEPS = new Set([
+  '@babel/core',
+  '@babel/runtime', 
+  '@babel/template',
+  '@babel/parser',
+  'babel-core',
+  'babel-runtime',
+  'babel-template',
+  'babylon'
+]);
+
+async function inferBabelVersion(asset, babelrc) {
+  // Attempt to determine version based on dependencies of plugins
+  let basedir = path.dirname(asset.name);
+  let presets = (babelrc.presets || []).map(p => resolveModule('preset', getPluginName(p), basedir));
+  let plugins = (babelrc.plugins || []).map(p => resolveModule('plugin', getPluginName(p), basedir));
+  let results = await Promise.all([...presets, ...plugins]);
+  let version;
+
+  for (let pkg of results) {
+    if (!pkg) {
+      continue;
+    }
+
+    for (let name of CORE_DEPS) {
+      let dep = getDependency(pkg, name);
+      if (dep) {
+        // Parse version range (ignore prerelease), and ensure it overlaps with the existing version (if any)
+        let range = new semver.Range(dep.replace(/-.*(\s|\|\||$)?/, ''));
+        if (version && !version.intersects(range)) {
+          throw new Error('Conflicting babel versions found in .babelrc. Make sure all of your plugins and presets depend on the same major version of babel.');
+        }
+
+        version = range;
+        break;
+      }
+    }
+  }
+
+  // Find the maximum major version allowed in the range and use that.
+  // e.g. if ^6 || ^7 were specified, use 7.
+  version = getMaxMajor(version);
+  if (!version) {
+    logger.warn(`Could not infer babel version. Defaulting to babel 7. Please add either babel-core or @babel/core as a dependency.`);
+    version = 7;
+  }
+
+  return version;
+}
+
+function getPluginName(p) {
+  return Array.isArray(p) ? p[0] : p;
+}
+
+function getMaxMajor(version) {
+  try {
+    let range = new semver.Range(version);
+    let sorted = range.set.sort((a, b) => a[0].semver.compare(b[0].semver));
+    return semver.major(sorted.pop()[0].semver.version);
+  } catch (err) {
+    return null;
+  }
+}
+
+async function resolveModule(type, name, basedir) {
+  try {
+    name = standardizeName(type, name);
+    return await resolve(name, {basedir}).then(([name, pkg]) => pkg);
+  } catch (err) {
+    return null;
+  }
+}
+
+// Copied from https://github.com/babel/babel/blob/3a399d1eb907df520f2b85bf9ddbc6533e256f6d/packages/babel-core/src/config/files/plugins.js#L61
+
+const EXACT_RE = /^module:/;
+const BABEL_PLUGIN_PREFIX_RE = /^(?!@|module:|[^/]+\/|babel-plugin-)/;
+const BABEL_PRESET_PREFIX_RE = /^(?!@|module:|[^/]+\/|babel-preset-)/;
+const BABEL_PLUGIN_ORG_RE = /^(@babel\/)(?!plugin-|[^/]+\/)/;
+const BABEL_PRESET_ORG_RE = /^(@babel\/)(?!preset-|[^/]+\/)/;
+const OTHER_PLUGIN_ORG_RE = /^(@(?!babel\/)[^/]+\/)(?![^/]*babel-plugin(?:-|\/|$)|[^/]+\/)/;
+const OTHER_PRESET_ORG_RE = /^(@(?!babel\/)[^/]+\/)(?![^/]*babel-preset(?:-|\/|$)|[^/]+\/)/;
+const OTHER_ORG_DEFAULT_RE = /^(@(?!babel$)[^/]+)$/;
+
+function standardizeName(type, name) {
```

Comment:
would be cool if babel could expose this function as a util

Reply:
This is used to infer the version of babel to use based on the plugins. I need to resolve each plugin's package.json to check its dependency on babel-core.